src/volume_mgr.rs

```diff
@@ -103,7 +103,36 @@ where
         &mut self,
         volume_idx: VolumeIdx,
     ) -> Result<Volume<D, T, MAX_DIRS, MAX_FILES, MAX_VOLUMES>, Error<D::Error>> {
-        let v = self.open_raw_volume(volume_idx)?;
+        return self._open_volume(volume_idx, false);
+    }
+
+    /// Get a read only volume (or partition) based on entries in the Master Boot Record.
+    /// Opening and closing a read only volume is faster than a writable volume.
+    ///
+    /// We do not support GUID Partition Table disks. Nor do we support any
+    /// concept of drive letters - that is for a higher layer to handle.
+    pub fn open_volume_read_only(
+        &mut self,
+        volume_idx: VolumeIdx,
+    ) -> Result<Volume<D, T, MAX_DIRS, MAX_FILES, MAX_VOLUMES>, Error<D::Error>> {
+        return self._open_volume(volume_idx, true);
+    }
+
+    /// Get a volume (or partition) based on entries in the Master Boot Record.
+    ///
+    /// We do not support GUID Partition Table disks. Nor do we support any
+    /// concept of drive letters - that is for a higher layer to handle.
+    fn _open_volume(
+        &mut self,
+        volume_idx: VolumeIdx,
+        read_only: bool,
+    ) -> Result<Volume<D, T, MAX_DIRS, MAX_FILES, MAX_VOLUMES>, Error<D::Error>> {
+        let v = self.open_raw_volume(volume_idx, read_only)?;
+        if !read_only {
+            let idx = self.get_volume_by_id(v)?;
+            let VolumeType::Fat(volume_type) = &self.open_volumes[idx].volume_type;
+            self.set_volume_status_dirty(volume_type, true)?;
```

Comment:
I don't think you need to mark it as dirty until the first write occurs?

Reply:
Ah, I read the comment thread from the kernel. Yeah, I can see the value in marking it dirty on first mount.

Note to self - the Neotron OS will need changing to ensure the volume is unmounted when the last file is closed.

Reply:
But this flag does not fix my problem haha, I have to update somehow the fat, at least Linux does not check the fat and repairs it if needed. I thought when the bit is cleared (is dirty) during mount it will be "repaired" and the FAT table gets update, but this is not the case

Reply:
I don't think Linux will run an `fsck` on a block device unless you ask it to. It is Windows that prompts to run Scandisk if it things the volume wasn't cleanly shut down.